src/megameklab/com/printing/PrintInfantry.java

```diff
@@ -0,0 +1,440 @@
+
+import megamek.common.*;
+import megamek.common.options.IOption;
+import megamek.common.weapons.artillery.ArtilleryWeapon;
+import megamek.common.weapons.infantry.InfantryWeapon;
+import org.apache.batik.util.SVGConstants;
+import org.w3c.dom.Element;
+import org.w3c.dom.svg.SVGRectElement;
+
+import java.util.Enumeration;
+import java.util.StringJoiner;
+
+/**
+ * Lays out a record sheet block for a single infantry unit
+ */
+public class PrintInfantry extends PrintEntity {
+
+    private final Infantry infantry;
+
+    /**
+     * Creates an SVG object for the record sheet
+     *
+     * @param infantry The infantry to print
+     * @param startPage The print job page number for this sheet
+     * @param options Overrides the global options for which elements are printed
+     */
+    public PrintInfantry(Infantry infantry, int startPage, RecordSheetOptions options) {
+        super(startPage, options);
+        this.infantry = infantry;
+    }
+
+    /**
+     * Creates an SVG object for the record sheet using the global printing options
+     *
+     * @param infantry The infantry to print
+     * @param startPage The print job page number for this sheet
+     */
+    public PrintInfantry(Infantry infantry, int startPage) {
+        this(infantry, startPage, new RecordSheetOptions());
+    }
+
+    @Override
+    protected String getSVGFileName(int pageNumber) {
+        return "conventional_infantry_platoon.svg";
+    }
+
+
+    @Override
+    protected Entity getEntity() {
+        return infantry;
+    }
+
+    @Override
+    protected boolean isCenterlineLocation(int loc) {
+        return false;
+    }
+
+    @Override
+    protected String getRecordSheetTitle() {
+        // Not used
+        return "";
+    }
+
+    @Override
+    protected void writeTextFields() {
+        super.writeTextFields();
+
+        if (!infantry.canMakeAntiMekAttacks()) {
+            setTextField(PILOTING_SKILL + "0", "N/A");
+        }
+        writeFieldGuns();
+
+        for (int j = 1; j <= 30; j++) {
+            if (j > infantry.getShootingStrength()) {
+                hideElement(SOLDIER + j, true);
+                hideElement(NO_SOLDIER + j, false);
+            } else {
+                setTextField(DAMAGE + j, (int)Math.round(infantry.getDamagePerTrooper() * j));
+            }
+        }
+        InfantryWeapon rangeWeapon = infantry.getPrimaryWeapon();
+        if (infantry.getSecondaryWeapon() != null && infantry.getSecondaryN() > 1
+                && !infantry.getSecondaryWeapon().hasFlag(WeaponType.F_TAG)) {
+            rangeWeapon = infantry.getSecondaryWeapon();
+        }
+        boolean scuba = infantry.getMovementMode() == EntityMovementMode.INF_UMU
+                || infantry.getMovementMode() == EntityMovementMode.SUBMARINE;
+        hideElement(UW_LABEL, !scuba);
+        InfantryWeapon singleSecondary = (infantry.getSecondaryN() == 1)? infantry.getSecondaryWeapon() : null;
+        for (int j = 0; j <= 21; j++) {
```

Comment:
The soldiers themselves are 1-based and ranges are 0-based?

Reply:
Yes, which could be a little confusing but it coincides with the text label. The soldiers and numbered 1-30 and the range table headers start at zero.